src/net.cpp

```diff
@@ -462,19 +462,22 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
     if (pszDest) {
         const std::vector<CService> resolved{Lookup(pszDest, default_port, fNameLookup && !HaveNameProxy(), 256)};
         if (!resolved.empty()) {
-            const CService& rnd{resolved[GetRand(resolved.size())]};
```

Comment:
The old code would pick a random address from the list of resolved addresses. The new code would pick the last one. I think it is good to preserve the randomness. One way to achieve that would be to shuffle the result before iterating and picking the last one.

```cpp
if (!resolved.empty()) {
    Shuffle(resolved.begin(), resolved.end(), FastRandomContext{});
    for (const auto& r : resolved) {
```

or if the shuffle is perceived expensive for big lists, before the loop `auto to_pick = GetRand(resolved.size())` and inside the loop only assign `addrConnect` for the `to_pick`th element.

Reply:
I think `Shuffle` should do, given this picks on a list of resolved DNS lookups. Normally, that should be rather short, and if [this](https://tmblr.co/ZCK6wx1dJN3aZ) is still up to date, the max length should be 4095.